### DISCUSSION TITLE
Where exactly comes the block_no from?

### DISCUSSION THREAD

**Discussion by JawaClass** [pymupdf/PyMuPDF#2549] [Looking for help]
In the documentation is only stated every image gets a block and a block_no assigned. But how about Text Blocks? Where does the deicision come from counts as a block? 

Is a page segmentation performed or does block information stem from the time when the pdf processor created the PDF from eg. Word TextFields?

THX

**Comment by JorjMcKie** [comment]
All items (text and images) are always extracted in the sequence as they are stored in the pages contents object(s). This is also the sequence by which all PDF viewers process these objects.
Block numbers are simply the running number of each block (image or text). 
When preparing the output of page.get_text(), PyMuPDF itself accesses the content of an internal object (TextPage) prepared by MuPDF. If your extraction flags have excluded images, then PyMuPDF cannot know whether there are images on the page and accordingly will assign / count only text blocks.
For performance reasons, "text" and "words" extractions ignore images via flags by default. So if you need consistent block numbers across "dict", "words", "text", etc. you must use the same flags value.
